lib/util/legacy_dataset_sync.rb

```diff
@@ -55,4 +54,8 @@ def fetch_json(url)
     @logger.error "Failed to make the request to #{url}"
     return nil
   end
+
+  def there_are(datasets)
```

Comment:
when a method returns a boolean, it's idiomatic in Ruby to suffix the method with `?`, i.e. `there_are?(datasets)`.

Also, in this case, no need to check for the presence of datasets. If there aren't any, the block never gets called.

```
irb> [].each { |d| puts "hello" }
irb> []
```

Reply:
Possibly even better to return the value and do a standard `any?` on the result

Reply:
Good point re: the '?' - will change

Interesting re: the 'each' block

I think it still makes sense to check the json structure is correct no?

i.e I want to check that I have a 'results' response before proceeding ...

```
def there_are?(datasets)
    datsets['results'] && datasets['result']['results']
end
```

That way it won't blow up if I get something unexpected ...
```
datasets['foo']['bar']
```

Reply:
Changed to the below